orderer/consensus/etcdraft/consenter.go

```diff
@@ -103,6 +104,49 @@ func (c *Consenter) ReceiverByChain(channelID string) MessageReceiver {
 	return nil
 }
 
+// ValidateJoinBlock implements the JoinBlockValidator interface and validates the etcdraft metadata
+// in the join block passed to the channel participation API. It performs checks to ensure the orderer
+// won't crash when calling the consenter's HandleChain.
+// It also determines whether the node is a member of the consenter set defined in the join block.
+func (c *Consenter) ValidateJoinBlock(configBlock *common.Block) (isClusterMember bool, err error) {
+	// Fetch consenter metadata
+	metadata, err := protoutil.GetConsenterMetadataFromBlock(configBlock)
+	if err != nil {
+		return false, errors.Wrap(err, "failed to get consenter metadata from config block")
+	}
```

Comment:
The metadata can be nil (migration or genesis blocks) 

https://github.com/hyperledger/fabric/blob/5fd497e7bb33987b1f029f0a2d51c9aa9f277e10/orderer/consensus/etcdraft/consenter.go#L141

can add a test to check that?

Reply:
Done